### PR TITLE
add ExecuteFetchAsDba in tabletmanager server

### PR DESCRIPTION
**Pull request by yaoshengzhe**
1. Add ExecuteFetchAsDba api in tabletmanager server.
2. Rename the existing ExecuteFetch to ExecuteFetchAsApp.
3. ExecuteFetchAsDba creates a dba connection on demand and takes care
   of enable/disable binlog and reload schema.
4. Add GetDbaConnection func in MysqlDaemon interface.
5. make sure fakesqldb package always store queries in lower case.

### CODE REVIEW

**Comment by yaoshengzhe** [comment]
add one more commit in this PR: add DbName param to ExecuteFetchAsDba

ExecuteFetchAsDba will execute use database statement if given
dbName is not empty.

File: go/vt/mysqlctl/mysql_daemon.go [hunk 1]
```diff
@@ -284,11 +287,6 @@ func (fmd *FakeMysqlDaemon) GetSchema(dbName string, tables, excludeTables []str
 // GetDbConnection is part of the MysqlDaemon interface
 func (fmd *FakeMysqlDaemon) GetDbConnection(dbconfigName dbconfigs.DbConfigName) (dbconnpool.PoolConnection, error) {
```
Comment (alainjobart) [inline go/vt/mysqlctl/mysql_daemon.go]: This probably doesn't need the DbConfigName any more... can be gone from the API in the other files too.

File: go/vt/tabletmanager/agent_rpc_actions.go [hunk 2]
```diff
@@ -294,9 +302,24 @@ func (agent *ActionAgent) ExecuteFetch(ctx context.Context, query string, maxrow
...
 
+	if err == nil && reloadSchema {
+		agent.QueryServiceControl.ReloadSchema()
+	}
 	return qr, err
 }
 
+// ExecuteFetchAsApp will execute the given query, possibly disabling binlogs.
+// Should be called under RPCWrap.
+func (agent *ActionAgent) ExecuteFetchAsApp(ctx context.Context, query string, maxrows int, wantFields bool, dbconfigName dbconfigs.DbConfigName) (*proto.QueryResult, error) {
```
Comment (alainjobart) [inline go/vt/tabletmanager/agent_rpc_actions.go]: Same as before, no need for dbconfigName here.

**Comment by yaoshengzhe** [comment]
Done. Removed dbconfigName param in ExecuteFetchAsApp and GetDbConnection.

**Comment by alainjobart** [comment]
LGTM

**Pull request merged**